src/compiler/core.ts

```diff
@@ -171,6 +171,20 @@ namespace ts {
         return result;
     }
 
+    export function mapObject<T, U>(object: Map<T>, f: (key: string, x: T) => [string, U]): Map<U> {

```

Comment:
jsdoc would be nice here, specifically calling out what happens when keys are mapped to the same key (looks like last one wins and earlier ones are discarded)